### PR TITLE
Pull down Elements via npm

### PR DESCRIPTION
**Pull request by daniel-ac-martin**
Replaces the old copy & pasted govuk-elements with the new npm module.

Addresses #34.

This uses the most recent version of elements and could represent a breaking change depending on how old the replaced code was.

### CODE REVIEW

**Comment by daniel-ac-martin** [comment]
This PR passes the tests, but I have no idea how rigorous those are.

Is there a process for manually testing this module?

**Comment by gavboulton** [comment]
Elements hasn't been updated a long time; there will be a lot of breaking changes. The naming convention for panels and lists has changed. Here's the changelog https://github.com/alphagov/govuk_elements/blob/master/CHANGELOG.md. This will need a major version bump.

The sass tests just make sure it compiles. Testing will be a case of putting the updated module into a project.

File: assets/stylesheets/app.scss [hunk 1]
```diff
@@ -4,15 +4,15 @@
...
-@import "govuk-elements/tables";
+@import "../../node_modules/govuk-elements-sass/public/sass/elements/helpers";
+@import "../../node_modules/govuk-elements-sass/public/sass/elements/reset";
+@import "../../node_modules/govuk-elements-sass/public/sass/elements/elements-typography";
+@import "../../node_modules/govuk-elements-sass/public/sass/elements/forms";
+@import "../../node_modules/govuk-elements-sass/public/sass/elements/buttons";
+@import "../../node_modules/govuk-elements-sass/public/sass/elements/details";
+@import "../../node_modules/govuk-elements-sass/public/sass/elements/lists";
+@import "../../node_modules/govuk-elements-sass/public/sass/elements/panels";
+@import "../../node_modules/govuk-elements-sass/public/sass/elements/tables";
```
Comment (gavboulton) [inline assets/stylesheets/app.scss]: This path is likely to change depending on the installation. You'll need to do something similar to working out the module path for govuk_frontend_toolkit in https://github.com/UKHomeOffice/passports-frontend-toolkit/blob/master/assets/stylesheets/mixins.scss.tpl. Have a look at https://github.com/UKHomeOffice/passports-frontend-toolkit/blob/master/build

**Comment by gavboulton** [comment]
govuk-elements-sass v1.2.0 has been released today!

File: assets/stylesheets/app.scss (hunk 1 above)
Comment (daniel-ac-martin) [inline assets/stylesheets/app.scss]: Ah good idea. I'll do that.

**Comment by daniel-ac-martin** [comment]
It's because they hate me!

**Comment by daniel-ac-martin** [comment]
@gavboulton: Updated.

**Comment by daniel-ac-martin** [comment]
I should probably add that there is more to elements than we are currently pulling down.

@gavboulton, @JoeChapman, @easternbloc: Is the aim with this module to pull in all of elements or just a subset of it?

I can very easily make a commit adding in the extra bits.

**Comment by daniel-ac-martin** [comment]
BTW, there are things in those 'extras' that I would like. Such as the grid layout stuff.

**Comment by gavboulton** [comment]
If there are things that you want then add them in.

**Comment by daniel-ac-martin** [comment]
@gavboulton: Elements does provide an example script for pulling it in (after the govuk-frontend-toolkit) so perhaps we should just follow that. Unless we are particularly worried about bloat?

**Comment by gavboulton** [comment]
Depends on the bloat, but if things are useful they should go in.

**Comment by daniel-ac-martin** [comment]
Okay. Well in that case perhaps we should leave the extra stuff for now. (I don't need it after all.)
If someone else wants it they could raise a new PR.

File: assets/stylesheets/mixins.scss.tpl [hunk 2]
```diff
@@ -14,3 +14,14 @@
...
+// govuk-elements-sass
+@import "{{{govuk-elements-path}}}/helpers";
+@import "{{{govuk-elements-path}}}/reset";
+@import "{{{govuk-elements-path}}}/elements-typography";
+@import "{{{govuk-elements-path}}}/forms";
+@import "{{{govuk-elements-path}}}/buttons";
+@import "{{{govuk-elements-path}}}/details";
+@import "{{{govuk-elements-path}}}/lists";
+@import "{{{govuk-elements-path}}}/panels";
+@import "{{{govuk-elements-path}}}/tables";
```
Comment (gavboulton) [inline assets/stylesheets/mixins.scss.tpl]: These aren't really mixins. We import the mixins as a standalone thing to use with custom styles.

Comment (daniel-ac-martin) [inline assets/stylesheets/mixins.scss.tpl]: Shall I just move them to their own file?

Comment (daniel-ac-martin) [inline assets/stylesheets/mixins.scss.tpl]: elements.scss.tpl ?

**Comment by daniel-ac-martin** [comment]
@gavboulton: Just updated. - What do you think?

**Comment by roc** [comment]
+1 

I'm currently shimming in new things from govuk elements (that new panel indent hotness m8 omg) waiting for an update here. People using the HOF pattern for making forms are all also bound by this waiting game, unless they get creative.

**Comment by daniel-ac-martin** [comment]
This PR should probably be updated to pull down v1.2.1 of elements. (Or that could be done as a separate PR if people are in a rush for this.

File: .gitignore [hunk 3]
```diff
@@ -1,4 +1,5 @@
 node_modules
 assets/stylesheets/*.css
+assets/stylesheets/elements.scss
```
Comment (gavboulton) [inline .gitignore]: 'assets/stylesheets/govuk-elements.scss'?

File: assets/stylesheets/app.scss [hunk 4]
```diff
@@ -4,15 +4,7 @@
...
-@import "govuk-elements/helpers";
-@import "govuk-elements/reset";
-@import "govuk-elements/elements-typography";
-@import "govuk-elements/forms";
-@import "govuk-elements/buttons";
-@import "govuk-elements/details";
-@import "govuk-elements/lists";
-@import "govuk-elements/panels";
-@import "govuk-elements/tables";
+@import "elements";
```
Comment (gavboulton) [inline assets/stylesheets/app.scss]: 'govuk-elements'?

Comment (gavboulton) [inline assets/stylesheets/app.scss]: And rename the file too!

Comment (daniel-ac-martin) [inline assets/stylesheets/app.scss]: Will do, just give me a minute.

**Comment by daniel-ac-martin** [comment]
@gavboulton: Added a commit to address your most recent comments. I haven't force pushed so you should be able to just review the last commit to check they address your concerns.

Once this PR is considered ready to be merged the extra commit should be squashed. (GitHub has a fancy new feature for this. - A drop down on the second, confirm merge button. Happy to do the actual merge myself if people have trouble finding it.)

(Or I can manually squash first if that is preferred.)

**Comment by gavboulton** [comment]
hmpo-frontend-toolkit@5.0.0 published.

Thanks @daniel-ac-martin.

**Pull request merged**